clion/src/main/kotlin/org/rust/clion/debugger/runconfig/RsCLionDebugRunnerUtils.kt

```diff
@@ -14,19 +13,25 @@ import com.jetbrains.cidr.cpp.toolchains.CPPToolchainsConfigurable
 import com.jetbrains.cidr.toolchains.OSType
 import org.rust.cargo.runconfig.BuildResult.ToolchainError
 import org.rust.cargo.runconfig.BuildResult.ToolchainError.*
+import org.rust.cargo.toolchain.RsToolchain
+import org.rust.cargo.toolchain.wsl.RsWslToolchain
 import org.rust.debugger.runconfig.RsDebugRunnerUtils.ERROR_MESSAGE_TITLE
 
 object RsCLionDebugRunnerUtils {
 
-    fun checkToolchainSupported(host: String): ToolchainError? {
+    fun checkToolchainSupported(toolchain: RsToolchain?, host: String): ToolchainError? {
         val toolSet = CPPToolchains.getInstance().defaultToolchain?.toolSet ?: return null
         if (CPPToolchains.getInstance().osType == OSType.WIN) {
-            val isMSVCRustToolchain = "msvc" in host
-            val isGNURustToolchain = "gnu" in host
+            if (toolchain is RsWslToolchain) {
+                return WSLWithNonWSL
+            }
```

Comment:
Note, if your current toolchain is WSL, `osType` is still `WIN` so the current code always shows error dialog even with proper setup